### DISCUSSION TITLE
Remove all listeners

### DISCUSSION THREAD

**Discussion by sammachin** [matter-js/matter.js#1461] [Feature requests (vote via thumbs up)]
I'd like to be able to call removeAllListeners() for an event without specfying a function and have it remove all listeners for that event.

**Comment by lauckhart** [comment]
Are you looking for a convenient way to remove listeners you've added? Or to disable listeners installed by default cluster implementations?

For the former ObserverGroup might be useful. It'll remove one or more listeners without specifying them explicitly

**Comment by sammachin** [comment]
yes its for removing listeners I added, any more notes on ObserverGroup?

**Comment by lauckhart** [comment]
Sure, it works like this:

```ts
import { ObserverGroup } from "@matter/main";

// create a device

const observers = new ObserverGroup();

observers.on(device.events.identify.startIdentifying, () => console.log("Started identifying"));
observers.on(device.events.identify.stopIdentifying, () => console.log("Stopped identifying"));

// do some stuff

observers.close();

// Listeners added above are removed
```
